docs/design.md

```diff
@@ -0,0 +1,153 @@
+# Design
+
+## Models
+
+A Model object represents a logical ML model. The object describes a location
+where ML model files can be stored for a specific model. 
+
+Optionally, a loader can be specified to load ML models weights and biases from specific sources.
+For example, a commonly used model loader would be the HuggingFace model loader
+where by providing the model name would load the model to the Model object
+bucket location. Model loaders are defined as container images and can either
+point to prebuilt container images or a Git repository with Dockerfile.
+
+A model
+
+The Model object can optionally describe the training parameters including a reference to the Dataset that should be used for training. In this case, the Model object represents the eventual output of a training/finetuning operation.
+
+```yaml
+kind: Model
+spec:
+  container: # used for either loader or trainer
+  loader: {} # Optional, when empty an just a location will be created with no files in it
+  trainer: {} # Optional, will run a training job
+```
+
+### Model Sources
+
+A Model can be sourced from Git or a prebuilt image. Using Git will trigger a container build Job.
+
+```yaml
+kind: Model
+name: falcon-7b
+spec:
+  container:
+    git:
+      url: https://github.com/substratusai/hf-model-loader
+    image: substratusai/hf-model-loader
+  loader:
+    params: {name: tiuu/falcon-7b}
```

Comment:
can we convert any param.x to PARAM_X as env variable?

So params.name would be available as env variable `PARAM_NAME` in the container itself.

What you think? @brandonjbjelland @nstogner

Reply:
https://github.com/substratusai/substratus/pull/75#discussion_r1257490669